controllers/operator/routes.go

```diff
@@ -321,22 +331,23 @@ func (r *AuthenticationReconciler) reconcileRoute(authCR *operatorv1alpha1.Authe
 
 func (r *AuthenticationReconciler) ensureRouteDoesNotExist(ctx context.Context, authCR *operatorv1alpha1.Authentication, fields *reconcileRouteFields) (result *ctrl.Result, err error) {
 	reqLogger := logf.FromContext(ctx)
-	reqLogger.Info("Determined Route should not exist; removing if present")
-	observedRoute := &routev1.Route{}
-	err = r.Get(ctx, types.NamespacedName{Name: fields.Name, Namespace: authCR.Namespace}, observedRoute)
-	if k8sErrors.IsNotFound(err) {
-		return subreconciler.ContinueReconciling()
-	} else if err != nil {
-		reqLogger.Error(err, "Failed to get existing route for reconciliation")
-		return subreconciler.RequeueWithError(err)
-	}
-	err = r.Delete(ctx, observedRoute)
-	if err != nil {
-		reqLogger.Error(err, "Failed to delete the Route")
-		return subreconciler.RequeueWithError(err)
+	if fields.Name != IMCrtAuthRouteName {
```

Comment:
I'd prefer inverting this check to `fields.Name == IMCrtAuthRouteName` and immediately returning with `subreconciler.ContinueReconciling()` when true - it reduces the necessary indentation and doesn't signal a requeue when no changes to cluster state have been committed.

That said, is this Route supposed to always exist, or are there situations when it won't? I was originally under the impression that this would only be here when mTLS was actually in use.

Reply:
@rwhundley , addressed your review comment. 
On the point 2, your thought is right. but currently it is always exist. As We don't know when Customer registers an identity provider configuration for certificate based authentication, Operator doesn't know unless it contacts the DB and check for this in the DB and it also doesn't know when to contact DB to see if cert based authentication is enabled or disabled. To avoid overhead of making Db calls we are creating this route by default.